pkg/logproto/sketch.proto

```diff
@@ -0,0 +1,30 @@
+syntax = "proto3";
+
+package logproto;
+
+option go_package = "github.com/grafana/loki/pkg/logproto";
+
+message CountMinSketch {
+  uint32 depth = 1;
+  uint32 width = 2;
+
+  // counters is a matrix of depth * width.
+  repeated uint32 counters = 3;
+}
+
+message TopK {
+  CountMinSketch cms = 1;
+  repeated string labels = 2;
+  bytes hyperloglog = 3;
+}
+
+message SketchSeries {
```

Comment:
Maybe name this something like `TopKMatrix`? There will be many different sketch types. Roughly using the same `Vector, Matrix` semantics that prometheus uses where `Vector` is a list of points for one timestamp and `Matrix` is a list of vectors.

Reply:
I like that. I was unsure about Matrix and Vector and didn't want to cause any confusion.